src/psyclone/tests/psyir/nodes/omp_directives_test.py

```diff
@@ -256,6 +282,86 @@ def test_ompdo_constructor():
     ompdo = OMPDoDirective(parent=schedule, children=[child])
     assert len(ompdo.dir_body.children) == 1
 
+    # Constructor with non-default parameters
+    ompdo = OMPDoDirective(omp_schedule="dynamic", collapse=4, reprod=True)
+    assert ompdo.omp_schedule == "dynamic"
+    assert ompdo.collapse == 4
+    assert ompdo.reprod
+    assert str(ompdo) == "OMPDoDirective[omp_schedule=dynamic,collapse=4]"
```

Comment:
`reprod` not mentioned in the str because it is not applicable to this directive?

Reply:
I did not change this behavior, although I can if you want.

Currently reprod is only printed if the self.reductions() method (a node method that looks at the properties of the Kern inside) returns anything. In this example there are no reduction kernels (I think only in LFRic there are)

Reply:
Oh I see. That's OK then.